### PR TITLE
Set schema when calling Database.connect()

### PR DESCRIPTION
**Pull request by hfazai**
This PR is to add the option of specifying default schema when calling `Database.connect()`

//Usage:

````Kotlin
Database.connect(url, user, password, schema = Schema("myschema"), setupConnection, manager)
````

OR

````Kotlin
Database.connect({ jdbcConnection }, schema = Schema("myschema"), setupConnection, manager)
````

OR

````Kotlin
Database.connect({ dataSource }, schema = Schema("myschema"), setupConnection, manager)
````

### CODE REVIEW

File: exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Database.kt [hunk 1]
```diff
@@ -76,50 +76,159 @@ class Database private constructor(private val resolvedVendor: String? = null, v
...
         ): Database {
-            return Database(explicitVendor) {
+            val database = Database(explicitVendor) {
                 connectionInstanceImpl(getNewConnection().apply { setupConnection(this) })
             }.apply {
                 TransactionManager.registerManager(this, manager(this))
             }
+
+            if(schema != null) {
+                with(database.transactionManager.newTransaction()) {
```
Comment (Tapac) [inline exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Database.kt:103]: I guess it should be replaced with `transaction` function or create transaction will not be closed ever.

**Review by Tapac — CHANGES_REQUESTED** [review]

File: exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Database.kt (hunk 1 above)
Comment (hfazai) [reply exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Database.kt]: I think Database.connect(schema = Schema("myschema")) could be calling outside a `transaction` and all next transactions should have the passed `schema` as the default schema. What do you think?

Comment (Tapac) [reply exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Database.kt]: I miss it but looks like `setScheme` will be called only once on some connection and all other connections will remain with their default scheme.

I think you have to call setScheme just after a new transaction was started.

Comment (hfazai) [reply exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Database.kt]: I did some changes but I'm not sure if this is a correct behavior

````KOTLIN
connect(schema = myschema)

transaction {
   // Default schema : myschema
}
````

// OR :

````KOTLIN

connect()
transaction { 
// Default schema : PUBLIC
 }

transaction {
  connect(schema = myschema)
  transaction {
    // Default schema : myschema
  }
  // Default schema : myschema
}
transaction {
   // Default schema : myschema
}
````

Comment (hfazai) [reply exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Database.kt]: @Tapac as you mentioned `setSchema` will change schema for some connection. So to do that you can simply call `SchemaUtils.setSchema()` inside a transaction. So I think the extension in this PR becomes useless. Do you agree?

**Pull request closed without merging**